src/restrictededitingmodeediting.js

```diff
@@ -386,3 +362,74 @@ function isRangeInsideSingleMarker( editor, range ) {
 
 	return markerAtStart && markerAtEnd && markerAtEnd === markerAtStart;
 }
+
+// Returns a post-fixer that reverts attributes changes made outside exception markers.
+function attributeOutsideExceptionMarkersPostFixer( editor ) {
+	return writer => {
+		let changeApplied = false;
+
+		for ( const change of editor.model.document.differ.getChanges() ) {
+			if ( change.type === 'attribute' ) {
+				const { range, attributeKey, attributeOldValue } = change;
+
+				if ( !isRangeInsideSingleMarker( editor, range ) ) {
+					const markersRanges = Array.from( editor.model.markers.getMarkersIntersectingRange( range ) )
+						.map( marker => marker.getRange() );
+
+					const rangesToRevert = getMultiRangeDifference( range, markersRanges );
+
+					for ( const range of rangesToRevert ) {
+						writer.setAttribute( attributeKey, attributeOldValue, range );
+
+						changeApplied = true;
+					}
+				}
+			}
+		}
+
+		return changeApplied;
+	};
+}
+
+// Calculates a difference of one range with many ranges (exclusion).
```

Comment:
We already have such function in track changes :(. I guess it is time to move it to `model.Range`.